tests/models/test_restore.py

```diff
@@ -435,7 +435,7 @@ def on_train_start(self):
             assert self.trainer.current_epoch == real_global_epoch and self.trainer.current_epoch > 0
 
             dataloader = self.train_dataloader()
-            tpipes.run_prediction(self.trainer.model, dataloader, dp=True)
+            tpipes.run_prediction(self.trainer.get_model(), dataloader)
```

Comment:
@rohitgr7 this was the actual fix for the dp stuff, I reverted the other dp specific changes in the run_prediction function. 
the run prediction function does not need the dp wrapped model.

Reply:
I just updated it according to the EvalModel before. but yeah this works too.